enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,167 @@
+- In order to support provider networks in IPI, the subnets passed to the installer must meet these requirements:
+  - have the capacity and ability for the installer to provision ports on the nodes subnet
+  - have dhcp enabled
+  - the installer must be able to send and recieve https traffic to nodes connected to ports on the nodes subnet
+- In order to support provider networks in UPI, additional documentation will be provided to support this use case
+- For those who do not want the installer to provision their networking resources, we will provide additional UPI documentation.
+- The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+- Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+
+## User Stories
+1. As an administrator of a private openstack cloud, I use provider networks for my openstack cluster, and would like to be able to install OpenShift in this environment.
+2. As an OpenStack tenant, I want to be able to use pre-configured network resources so that I dont have to create new resources to install an OpenShift cluster.
+
+## Implementation Details Notes/Constraints
+
+### Installer Resources
+If a user wants to use a pre-existing subnet, they can provide the installer with the ID of that subnet. The installer needs to have the authorization and quota to provision the necessary ports and resources on that subnet.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    nodesSubnet: abcd-4321
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+Users with provider networks often do not use floating IPs, and so the first changge we will make to address this is to make the parameter `platform.openstack.lbFloatingIP` optional. When unset, the installer will not attempt to provision and attach a floating ip to the api port. To allow the customer to set up their own custom external access infrastructure, we will have to let them set the `apiVIP` and `ingressVIP` port IPs ahead of the install, so that they can be added to their routing/loadbalancing scheme before the installer is run. 
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    apiVIP: "192.168.30.15"
+    ingressVIP: "192.168.30.17"
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+Lastly, the installer will need to know how to access the api, so an optional argument, `apiEndpoint`, that takes an IP or URL as an argument will be added. When this argument is set, the installer will direct all api calls to the address the user provides.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    apiEndpoint: "http://anyURLorIP.com"
```

Comment:
The API will always listens on port 6443 served via HTTPS. Or is the idea to put a proxy in front of it?

Reply:
Yeah a proxy or a loadbalancer/wildcard

Reply:
The idea is to enable cluster administrators to rout traffic in and out of the cluster however they want. So if they want to create the dns entries before running the installer, if the want to alias the api to another domain name, or if they want to set up a proxy it still works. This lets the installer reach the api the same way they intend to set it up for customers/users so they don't have to do any additional setup work post install.